### PR TITLE
testing add address sanitizer runs

### PR DESCRIPTION
**Pull request by burlen**
this will catch memory leaks, invalid memory read/write, etc...

let's get #27 in before merging this

### CODE REVIEW

**Comment by rileyjmurray** [comment]
Ping @burlen: I resolved a merge conflict so that we could run CI. Not sure if I resolved in the appropriate way, but we're getting a CMake error with these changes.

File: RandBLAS/CMakeLists.txt [hunk 1]
```diff
@@ -17,8 +17,15 @@ target_include_directories(RandBLAS INTERFACE
...
-target_link_libraries(RandBLAS INTERFACE ${RandBLAS_libs})
-target_compile_options(RandBLAS INTERFACE -Wall -Wextra)
+target_link_libraries(RandBLAS PUBLIC ${RandBLAS_libs})
+
+set(RandBLAS_cxx_opts -Wall -Wextra)
+if (SANITIZE_ADDRESS)
+    list(APPEND RandBLAS_cxx_opts -fsanitize=address)
+    target_link_options(RandBLAS PUBLIC -fsanitize=address)
+endif()
+
```
Comment (rileyjmurray) [inline RandBLAS/CMakeLists.txt:27]: @burlen  CMake is angry that the ``INTERFACE`` specifications in ``target_link_libraries`` and ``target_compile_options`` has been replaced with ``PUBLIC``. It looks like changing these back to ``INTERFACE`` fixes things.

**Comment by rileyjmurray** [comment]
I fixed a memory leak in ``test_sparse.cc`` identified by the address sanitizer. However, address sanitizer also identified nontrivial leaks in ``dense.hh``.

File: RandBLAS/CMakeLists.txt (hunk 1 above)
Comment (burlen) [reply RandBLAS/CMakeLists.txt]: yes, that was because of the change to header only (use INTERFACE) from a compiled library(use PUBLIC)

**Comment by burlen** [comment]
@rileyjmurray leak fixed. macos failure is due to test tolerance not met

**Review by rileyjmurray — APPROVED** [review]

**Pull request merged**